### PR TITLE
Update to discord.js v14 

### PR DESCRIPTION
**Pull request by jmiln**
This updates stuff to be discord.js v14 compatible, which was needed in order to update discord-player, which in turn was needed to make Spotify links work again apparently.

- This probably did screw up some indenting, my environment is set for 4 spaces instead of tabs (Whoops)

### CODE REVIEW

**Comment by whirlxd** [comment]
Jesus 
Thanks homie , ill take a look over it next week and either merge it or sideline the v 13 branch and make this the main branch

**Comment by jmiln** [comment]
Sounds good. Thanks for the solid bot. This was the first music bot I've really worked with, and it's all nicely laid out and easy to make sense of

**Comment by whirlxd** [comment]
aight i got sick for a few days so couldn't review it , will do so

**Review by virtualized-human — CHANGES_REQUESTED** [review]
In files:

commands\context-menus\message\add.js
commands\slash\music\autoplay.js
commands\slash\music\play.js
commands\slash\music\repeat.js

image

there are "discord--player" which makes the bot crashing. please change it to "discord-player", otherwise the bot dont start:

`D:\Documents\Testing\Musico-musico>npm run musico

> musico@3.0.1 musico
> node musico.js

node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module 'discord--player'
Require stack:
- D:\Documents\Testing\Musico-musico\commands\context-menus\message\add.js
- D:\Documents\Testing\Musico-musico\handlers\interactionHandlers.js
- D:\Documents\Testing\Musico-musico\musico.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (D:\Documents\Testing\Musico-musico\commands\context-menus\message\add.js:1:23)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\Documents\\Testing\\Musico-musico\\commands\\context-menus\\message\\add.js',
    'D:\\Documents\\Testing\\Musico-musico\\handlers\\interactionHandlers.js',
    'D:\\Documents\\Testing\\Musico-musico\\musico.js'
  ]
}

D:\Documents\Testing\Musico-musico>npm run musico

> musico@3.0.1 musico
> node musico.js

[Musico] | Logged in as Musicio#3682!
[Musico] | Servers! ["0"]
[Musico] | Users! ["1"]
[Musico] | Channels! ["0"]
[Musico] | Loaded all (/) undefined commands

D:\Documents\Testing\Musico-musico>`

File: commands/context-menus/message/add.js [hunk 1]
```diff
@@ -1,87 +1,92 @@
-const { QueryType } = require("discord-player");
-const { MessageEmbed } = require("discord.js");
+const { QueryType } = require("discord--player");
```
Comment (virtualized-human) [inline commands/context-menus/message/add.js]: Change from "discord--player" to "discord-player"

File: commands/slash/music/autoplay.js [hunk 2]
```diff
@@ -1,66 +1,66 @@
 const { SlashCommandBuilder } = require("@discordjs/builders");
-const { QueueRepeatMode } = require("discord-player");
+const { QueueRepeatMode } = require("discord--player");
```
Comment (virtualized-human) [inline commands/slash/music/autoplay.js]: Change from "discord--player" to "discord-player"

File: commands/slash/music/play.js [hunk 3]
```diff
@@ -1,92 +1,99 @@
 // Deconstructed the constants we need in this file.
 
-const { MessageEmbed } = require("discord.js");
-const { QueryType } = require("discord-player");
+const { EmbedBuilder: MessageEmbed } = require("discord.js");
+const { QueryType } = require("discord--player");
```
Comment (virtualized-human) [inline commands/slash/music/play.js]: Change from "discord--player" to "discord-player"

File: commands/slash/music/repeat.js [hunk 4]
```diff
@@ -1,72 +1,78 @@
 const { SlashCommandBuilder } = require("@discordjs/builders");
-const { QueueRepeatMode } = require("discord-player");
+const { QueueRepeatMode } = require("discord--player");
```
Comment (virtualized-human) [inline commands/slash/music/repeat.js]: Change from "discord--player" to "discord-player"

**Review by virtualized-human — CHANGES_REQUESTED** [review]

**Comment by jmiln** [comment]
Those are fixed, thanks for catching those. 

I'd switched it to discord-portable-player for a while since it was basically the same but had updated to Discord.js v13 first, and I guess I only removed the portable part, forgetting a dash

**Comment by virtualized-human** [comment]
@Whirl21 dont forget it!

**Review by whirlxd — APPROVED** [review]
aights

**Pull request merged**